relayer/chains/cosmos/tx.go

```diff
@@ -213,8 +213,11 @@ func (cc *CosmosProvider) broadcastTx(
 		cc.LogFailedTx(rlyResp, err, msgs)
 		return err
 	}
-
-	cc.UpdateFeesSpent(cc.ChainId(), cc.Key(), fees)
+	address, err := cc.Address()
+	if err != nil {
+		return fmt.Errorf("Failed to get relayer bech32 wallet addresss, %w", err)
+	}
+	cc.UpdateFeesSpent(cc.ChainId(), cc.Key(), address, fees)
```

Comment:
Are we sure it's safe to return here if we can't get the address?

This changes the control flow. If we error here, then line 225 `go cc.waitForTx(asyncCtx, res.Hash, msgs, asyncTimeout, asyncCallback)` never gets called. 

This differs from the previous control flow that will always reach line 225.

We could log an error and continue as normal; only recording the metric if we get an address.

Reply:
I think the previous control flow does return an error prior to 225? line: 199 and 214.
Maybe I'm overlooking something.

Regardless, I like your idea of simply logging the the error rather than returning.

Reply:
Line 199 and 214 do return an error and exit appropriately. 

However, we're introducing a new error path if `cc.Address()` fails.

Reply:
Yes. I see. I mis-interpreted your first comment above. 
Thanks for the explanations @DavidNix